### ISSUE TITLE
How do we convey legend and description for screen-readers in `fieldset`

### ISSUE THREAD

**Issue by mimarz** [digdir/designsystem#4662]

**Comment by eirikbacker** [comment]
`<fieldset>` descriptions should be accessible to screen reader users. However, using `aria-describedby` on `<fieldset>` causes all child `<input>` elements to inherit the same description, resulting in redundant and confusing announcements. To avoid this, we use `aria-labelledby` to reference both the legend and the description. `aria-labelledby` is only announced when screen readers enter the `fieldset`, not when navigating its child elements. This means the accessible name of `<fieldset>` includes both the legend and description, which may differ from some test expectations, but as of March 2026, this approach provides the best user experience across assistive technologies. This approach is also verified by the chief of accessibility at NRK and the accessibility expert at NAV.
